homework/index.js

```diff
@@ -30,13 +30,103 @@
     return elem;
   }
 
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const jsonString = JSON.stringify(data, null, 2);
```

Comment:
I didn't get that, Jim. When I tries to get rid one of them, data isnt shown in my web page, select element doesnt work I mean.

Reply:
The `data` parameter in the line `fetchJSON(url, (err, data) => {` is already the array of repository objects that you need (console.log it if necessary). The extra step of `JSON.stringify` and `JSON.parse` is unneeded (and therefore should be removed). Since you used the name `repositories` to store the result of the `JSON.parse`, you could just rename the `data` parameter from `fetchJSON` call to `repositories` and remove the lines containing `JSON.stringify` and `JSON.parse`.